packages/0/src/composables/useLayout/index.ts

```diff
@@ -63,64 +126,196 @@ export function useLayout<
+        for (const key in el) {
+          const candidate = unref((el as any)[key])
+          if (candidate instanceof HTMLElement) {
+            return candidate[valueProp as keyof HTMLElement] ?? registrant.value ?? 0
+          }
+        }
+      }
+
+      return registrant.element?.value?.[valueProp] ?? registrant.value
+    })
+
+    const ticket = registry.register({
+      ...registrant,
+      position: registrant.position!,
+      order: registrant.order ?? 0,
+      size,
+    }) as Z
+
+    const cumulative = computed(() => {
+      const offsets = { left: 0, right: 0, top: 0, bottom: 0 }
+      for (const current of registry.values()) {
+        if (!current.isSelected.value) continue
+        if (current.index >= ticket.index && (ticket.position !== opposites[current.position])) break
+        offsets[current.position] += unref(current.size) ?? unref(current.value)
+      }
 
-    const ticket = registry.register(item)
+      return {
+        ...offsets,
+        height: offsets.top + offsets.bottom,
+        width: offsets.left + offsets.right,
+      }
+    })
+    const height = computed(() => isVertical(ticket.position) ? unref(ticket.size ?? ticket.value) : bottom.value - top.value - cumulative.value.height)
+    const width = computed(() => isVertical(ticket.position) ? right.value - left.value - cumulative.value.width : unref(ticket.size ?? ticket.value))
+    const x = computed(() => (
+      ticket.position === 'left'
+        ? left.value + cumulative.value.left
+        : (
+            ticket.position === 'right'
+              ? right.value - width.value - cumulative.value.right
+              : left.value + cumulative.value.left
+          )
+    ))
+    const y = computed(() => (
+      ticket.position === 'top'
+        ? top.value + cumulative.value.top
+        : (
+            ticket.position === 'bottom'
+              ? bottom.value - height.value - cumulative.value.bottom
+              : top.value + cumulative.value.top
+          )
+    ))
 
-    sizes.set(ticket.id, ticket.value)
+    Object.assign(ticket, {
+      cumulative,
+      x,
+      y,
+      width,
+      height,
+      rect: { x, y, width, height },
+    })
 
     return ticket
   }
 
-  if (IN_BROWSER && getCurrentInstance()) {
-    function resize () {
-      height.value = window.innerHeight
+  function resize () {
+    if (el?.value) {
+      const rect = el.value.getBoundingClientRect()
+      left.value = rect.left
+      top.value = rect.top
+      right.value = rect.left + rect.width
+      bottom.value = rect.top + rect.height
+      width.value = rect.width
+      height.value = rect.height
+    } else {
       width.value = window.innerWidth
+      height.value = window.innerHeight
+      left.value = 0
+      top.value = 0
+      right.value = window.innerWidth
+      bottom.value = window.innerHeight
     }
+  }
 
+  let observer: ResizeObserver | null = null
+
+  if (IN_BROWSER && getCurrentInstance()) {
```

Comment:
still using `IN_BROWSER`? this can break Nuxt SSR, we'll need to add some "resolver" for this

Reply:
if we just ran this onMounted would we run into weird hydration issues?